src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/EditorInProcess.cs

```diff
@@ -132,6 +142,242 @@ public async Task SetUseSuggestionModeAsync(bool value, CancellationToken cancel
+        {
+            await TestServices.Workspace.WaitForAsyncOperationsAsync(FeatureAttribute.NavigationBar, cancellationToken);
+
+            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            var view = await GetActiveTextViewAsync(cancellationToken);
+            var combobox = (await GetNavigationBarComboBoxesAsync(view, cancellationToken))[index];
+            return combobox.Items.OfType<object>().SelectAsArray(i => $"{i}");
+        }
+
+        public async Task<string?> GetProjectNavigationBarSelectionAsync(CancellationToken cancellationToken)
+        {
+            return await GetNavigationBarSelectionAsync(index: 0, cancellationToken);
+        }
+
+        public async Task<string?> GetTypeNavigationBarSelectionAsync(CancellationToken cancellationToken)
+        {
+            return await GetNavigationBarSelectionAsync(index: 1, cancellationToken);
+        }
+
+        public async Task<string?> GetMemberNavigationBarSelectionAsync(CancellationToken cancellationToken)
+        {
+            return await GetNavigationBarSelectionAsync(index: 2, cancellationToken);
+        }
+
+        public async Task<string?> GetNavigationBarSelectionAsync(int index, CancellationToken cancellationToken)
+        {
+            await TestServices.Workspace.WaitForAsyncOperationsAsync(FeatureAttribute.NavigationBar, cancellationToken);
+
+            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            var view = await GetActiveTextViewAsync(cancellationToken);
+            var combobox = (await GetNavigationBarComboBoxesAsync(view, cancellationToken))[index];
+            return combobox.SelectedItem?.ToString();
+        }
+
+        public async Task SelectProjectNavigationBarItemAsync(string item, CancellationToken cancellationToken)
+        {
+            await SelectNavigationBarItemAsync(index: 0, item, cancellationToken);
+        }
+
+        public async Task SelectTypeNavigationBarItemAsync(string item, CancellationToken cancellationToken)
+        {
+            await SelectNavigationBarItemAsync(index: 1, item, cancellationToken);
+        }
+
+        public async Task SelectMemberNavigationBarItemAsync(string item, CancellationToken cancellationToken)
+        {
+            await SelectNavigationBarItemAsync(index: 2, item, cancellationToken);
+        }
+
+        public async Task SelectNavigationBarItemAsync(int index, string item, CancellationToken cancellationToken)
+        {
+            await TestServices.Workspace.WaitForAsyncOperationsAsync(FeatureAttribute.NavigationBar, cancellationToken);
+
+            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            var itemIndex = await GetNavigationBarItemIndexAsync(index, item, cancellationToken);
+            if (itemIndex < 0)
+            {
+                throw new ArgumentException($"Could not find '{item}' in combobox");
```

Comment:
can you have this report the set of items that were there?

Reply:
➡️ Now using `Assert.Contains` on the failure path